lib/src/common_server_impl.dart

```diff
@@ -153,6 +154,13 @@ class CommonServerImpl {
   Future<proto.VersionResponse> version(proto.VersionRequest _) {
     final sdk = Sdk.create();
     final packageVersions = getPackageVersions(nullSafe: _nullSafety);
+    final packageInfos = [
+      for (var packageName in packageVersions.keys)
+        proto.PackageInfo()
+          ..name = packageName
+          ..version = packageVersions[packageName]!
+          ..supported = isSupportedPackage(packageName),
```

Comment:
A nit, but could you use `.entries` here to avoid the `!`?

```suggestion
      for (var package in packageVersions.entries)
        proto.PackageInfo()
          ..name = package.key
          ..version = package.value
          ..supported = isSupportedPackage(package.key),
```

Reply:
I hate losing the semantic meanings when using `entries`. :( I think it really is less readable coming back as a reader. "What is the 'key' of a package?" "What is the 'value' of a package?"

Reply:
That's fair and the `!` isn't an issue here. Thanks for the clarification of your thoughts :)